src/getDataFromTree.tsx

```diff
@@ -104,37 +105,34 @@ export type GetMarkupFromTreeOptions = {
 
 export function getMarkupFromTree({
   tree,
-  context = {},
   // The rendering function is configurable! We use renderToStaticMarkup as
   // the default, because it's a little less expensive than renderToString,
   // and legacy usage of getDataFromTree ignores the return value anyway.
   renderFunction = require("react-dom/server").renderToStaticMarkup,
 }: GetMarkupFromTreeOptions): Promise<string> {
   const renderPromises = new RenderPromises();
-
   class RenderPromisesProvider extends React.Component {
-    static childContextTypes: { [key: string]: any } = {
-      renderPromises: PropTypes.object,
-    };
-
-    getChildContext() {
-      return { ...context, renderPromises };
-    }
 
     render() {
       // Always re-render from the rootElement, even though it might seem
       // better to render the children of the component responsible for the
       // promise, because it is not possible to reconstruct the full context
       // of the original rendering (including all unknown context provider
       // elements) for a subtree of the orginal component tree.
-      return tree;
+      return (
+        <ApolloContext.Consumer>
+          {(value: any) => (
+            // TODO: this is where the issue lives, we provide a tree
```

Comment:
This is what i'm talking about. When tree starts with <ApolloProvider client={client}> it will kind off fail. 

This could be solved by adjusting the `ApolloProvider` analogue to this one, first consume a possibly existing context and then override the possibly existing context

Reply:
I implemented this, making it work on node env's but browser is still an issue.